enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,452 @@
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Upgradeable"
+  }
+]
+```
+
+- **Network operator** has below condition:
+
+```yaml
+[root@hv2 ~]# oc get co/network -ojson | jq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "ManagementStateDegraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "True",
+    "type": "Upgradeable"
+  },
+  {
+    "lastTransitionTime": "2022-06-07T15:35:25Z",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:37:57Z",
+    "status": "True",
+    "type": "Available"
+  }
+]
+```
+
+## Proposal
+
+This section proposes minimalistic changes without breaking the current design pattern. The below proposal/recommendation consists of more granular `status.conditions`  based on all the verification we already have in TALM.
+
+**Considering**: backup,precaching is true and enable for upgrade is true as well. 
+
+### Workflow Description
+
+1. Condition type related to cluster existing in the managedcluster. The type could reflect on the existence of the targeted cluster to be present in the managedcluster. 
+
+  ```yaml
+  Type:    "ClusterValidation/ClustersSelected",
+  Status:  True/False,
+  Reason:  "",
+  Message: “”
+  ```
+
+2. Condition type: **BackupDone**
+3. a) Spec validation for the `managed policies`. Currently it gets checked both during pre-caching and for upgrade. This check can be done before starting an upgrade, irrespective of pre-caching being true or false. b) Platform related verification. 
+Point a) and b) can be merged into PreUpgradeValidation where we check managed policies existence, conformity of platform image etc.
+
+ ```yaml
+  Type:    "PreUpgradeValidation",
+  Status:  True/False,
+  Reason:  "MissingManagedPolicy/InvalidPlatformImage/ValidationComplete",
+  Message: “”
+  ```
+
+4. Condition type: **PrecacheSpecValid** and **PrecachingDone** which is already in the current design. For **PrecacheSpecValid**, we may omit the highlighted Reason from the [current spec](#precachingSpec)
+5. Checking the blocking CRs status, if there is any issue UpgradeCannotStart reason is reported. If CRs are not blocked the upgrade can start. Also, adding and deleting labels to the cluster can be performed here. Condition type: Started/Progressing?
+
+ ```yaml
+    Type:    "Started/Progressing",
+    Status:  True/False,
+    Reason:  "MissingBlockingCR/IncompleteBlockingCR/MissingClusterLabel/InProgress",
+    Message: “”
+  ```
+
+  > Mohamed had a comment on this block: "In my opinion those are 2 different conditions 1) Initialized/Started means that the action (copying policies) is happening  2) Progressing/InProgress means that policy remediation started." To discuss.
```

Comment:
We should try to keep the conditions as high level/generic as we can without getting into how the operator is implemented. In this case I don't think we should have a condition on copying policies since that is part of the implementation (and we won't copy once Selective Policy Enforcement is available). That way we leave ourselves flexibility to change implementation as needed without being constrained by the conditions.

Reply:
@jc-rh @serngawy to discuss.

Reply:
Hi @imiller0, here do several other validation rather than only copying policies but we also build a plan and remediate it with policies. If we don't keep the condition, the next condition will later become the same old READY condition. In the CGU, if `enable: true`, we launch this condition and go through these validations, subsequently, the started condition become true and Reason is InProgress.